### PR TITLE
chore: fix checks

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

File: testplans/graphsync/main.go [hunk 1]
```diff
@@ -538,7 +538,7 @@ func runProvider(ctx context.Context, runenv *runtime.RunEnv, initCtx *run.InitC
 				// set up http server to send file
 				http.HandleFunc(fmt.Sprintf("/%s", node.Cid()), func(w http.ResponseWriter, r *http.Request) {
 					fileReader, err := os.Open(file.Name())
-					defer fileReader.Close()
+					defer func() { _ = fileReader.Close() }()
```
Comment (marten-seemann) [inline testplans/graphsync/main.go:541]: I don't think this is required, is it?

**Review by marten-seemann — APPROVED** [review]

File: testplans/graphsync/main.go (hunk 1 above)
Comment (Stebalien) [reply testplans/graphsync/main.go:541]: I got an error...

Comment (marten-seemann) [reply testplans/graphsync/main.go:541]: I see why: If `err != nil`, `fileReader` might be `nil`, and the `defer` will panic.
The correct fix is to move the `defer` after the error check (no `func()` required).

Comment (Stebalien) [reply testplans/graphsync/main.go:541]: Hm. You're right, but I'm not sure why we got the error we did (or why my fix worked).

https://github.com/ipfs/go-graphsync/pull/200

Comment (marten-seemann) [reply testplans/graphsync/main.go:541]: If that fix worked, it's probably a staticcheck (or was it vet?) bug.

**Pull request merged**